code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/DnsInfoDaoImpl.java

```diff
@@ -360,11 +360,17 @@ protected void addToResolve(ServiceInstanceData targetInstance, String aliasName
                 if (StringUtils.isEmpty(targetInstanceName)) {
                     ipToInstanceName.put(targetIp, null);
                 } else {
+                    String dnsName;
+                    if (targetInstance.getService().getKind().equalsIgnoreCase("kubernetesservice")) {
```

Comment:
@ibuildthecloud as this split just changes fqdn format from instanceName.globalNamespace (followed by regular service instances) to instanceName.serviceName.stackName.globalNamespace (followed by k8s service instances), so we can go with the label approach. The label would define whether svcName.stackName should be included to the individual container resolution.